realtime_tools/include/realtime_tools/realtime_publisher.hpp

```diff
@@ -78,6 +78,14 @@ class RealtimePublisher
   : publisher_(publisher), is_running_(false), keep_running_(true), turn_(State::LOOP_NOT_STARTED)
   {
     thread_ = std::thread(&RealtimePublisher::publishingLoop, this);
+
+    // Wait for the thread to be ready before proceeding
+    // This is important to ensure that the thread is properly initialized and ready to handle
+    // messages before any other operations are performed on the RealtimePublisher instance.
+    while (!thread_.joinable() ||
+           turn_.load(std::memory_order_acquire) != State::LOOP_NOT_STARTED) {
```

Comment:
Maybe I don't understand this but shouldn't it be
turn_.load(std::memory_order_acquire) == State::LOOP_NOT_STARTED

Reply:
You are right